consensus/beacon/consensus.go

```diff
@@ -366,7 +365,7 @@ func (beacon *Beacon) Finalize(chain consensus.ChainHeaderReader, header *types.
 		// fmt.Println("at block", header.Number, "performing transition?", state.Database().InTransition())
 		parent := chain.GetHeaderByHash(header.ParentHash)
 		if err := overlay.OverlayVerkleTransition(state, parent.Root, chain.Config().OverlayStride); err != nil {
-			log.Error("error performing the transition", "err", err)
+			panic(fmt.Sprintf("error performing the transition: %s", err))
```

Comment:
The current function doesn't return an error. But definitely we need to figure out that something failed in the conversion. 

Since `Finalize(...)` is an already existing method, I'm not sure we want to change its signature. If we can return an error LMK and I can bubble it up.

Reply:
I'll keep the log.Error before the panic.